core/storage/btree.rs

```diff
@@ -312,20 +381,43 @@ impl BTreeCursor {
                             "we don't iterate leaf cells while trying to move to a leaf cell"
                         );
                     }
-                    BTreeCell::IndexInteriorCell(_) => {
-                        unimplemented!();
+                    BTreeCell::IndexInteriorCell(IndexInteriorCell {
+                        left_child_page,
+                        payload,
+                        ..
+                    }) => {
+                        let SeekKey::IndexKey(index_key) = key else {
+                            unreachable!("index seek key should be a record");
+                        };
+                        let record = crate::storage::sqlite3_ondisk::read_record(payload)?;
+                        let comparison = match cmp {
```

Comment:
nit: can we rename comparison to found?

Reply:
https://github.com/penberg/limbo/pull/350/commits/8563d620afaffc74f2ab59e008a4bae5a4e32299

Renamed to `found` in `seek()` and to `target_leaf_page_is_in_left_subtree` in `move_to()`